mdsuite/utils/config.py

```diff
@@ -36,10 +38,22 @@ class Config:
     bokeh_sizing_mode: str
         The way bokeh scales plots.
         see bokeh / sizing_mode for more information
+    jupyter : bool
+            If true, jupyter is being used.
+    GPU: bool
+            TODO I think this is outdated.
+    memory_scaling_test : bool
+            If true, a scaling test is being performed and therefore, all batch sizes
+            are set to 1. Should typically be accompanied by the memory fraction being
+            set to 1 as well.
+    memory_fraction: bool
+            The portion of the available memory to be used.
     """
 
     jupyter: bool = False
     GPU: bool = False
+    memory_scaling_test: bool = False
+    memory_fraction: float = 0.5
```

Comment:
Maybe we could expand the config to be `config.memory.scaling_test = True` instead of `config.memory_scaling_test = True` with additional dataclasses. This way it could be more structured.

Reply:
Yeah that's an interesting idea. Managing how configuration things are set in general is a nice thing to discuss as it can be quite involved. I think having data classes for different things like you mention here would be very nice.